test/hooks.test.js

```diff
@@ -740,24 +740,25 @@ test('onRoute hook should able to change the route url', t => {
   })
 })
 
-test('onRoute hook that throws should be caught ', t => {
-  t.plan(1)
-  const fastify = Fastify()
-
-  fastify.register((instance, opts, done) => {
-    instance.addHook('onRoute', () => {
-      throw new Error('snap')
-    })
-    instance.get('/', opts, function (req, reply) {
-      reply.send()
-    })
-    done()
-  })
-
-  fastify.ready(err => {
-    t.ok(err)
-  })
-})
+// TODO: remove once review is done
```

Comment:
Now that also the `onRoute` hooks are sync, the error handling needs to also be sync. Just want to double-check and get feedback before modifying the test

Reply:
It's totally ok to handle the error synchronously